FreeRTOS/FreeRTOS
# Raspberry_Pi_Pico preliminary port.

Preliminary port of the Raspberry Pi Pico with GCC toolchain. 

Description
--------------
The demo is able to blink the onboard LED. Most of the code is
based on the Pi Pico SDK codebase.

Limitations
 - This port uses only Core 0 (Core 1 is always idle).
 - Most of the peripherals aren't readily usable as the module specific
   libraries haven't been ported yet.


Test Steps
-------------
Make sure that the GNU ARM embedded toolchain is on the path. Navigate to
FreeRTOS/Demo/CORTEX_M0+_RaspberryPiPico_GCC directory.
```
$ make
```
You should see: ```build/blinky.uf2```. This is the file that you need to drop into the FAT16 partition that appears when the Pico device is booted
into the USB device mode.

### Booting the Pi Pico into USB device mode
Press the BOOTSEL switch and plug in the Pi Pico to a USB cable attached to
the host computer. Now release the BOOTSEL switch.
You should see a mass storage device and its FAT16 partiton should be
automounted on the host computer.

### Flash the Pi Pico
Drag and drop the ```build/blinky.uf2``` file into the mounted partition.

-or-

copy it to the mounted partition (assuming it is mounted under /media/developer/RPI-RP2/):
```
cp build/blinky.uf2 /media/developer/RPI-RP2/
```

The Pi Pico will write into the flash with the relevant bits and reboot.

At this point you should see the onboard LED blinking.


Related Issue
-------------------
None

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice. However please maintain the original copyright notices from the files based on the Pico SDK.
